### PR TITLE
Add types for the eslint-ast-utils package

### PR DESCRIPTION
**Pull request by marksmccann**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Danger failed to run `dangerfile.ts`.
  </td>
    </tr>
  </tbody>
</table>



## Error Error
```
ENOENT: no such file or directory, open 'types/eslint-ast-utils/package.json'
Error: ENOENT: no such file or directory, open 'types/eslint-ast-utils/package.json'
    at Object.openSync (node:fs:590:3)
    at Object.readFileSync (node:fs:458:35)
    at dangerfile.ts:68:19
    at Array.forEach (<anonymous>)
    at Object.<anonymous> (dangerfile.ts:67:21)
    at Module._compile (node:internal/modules/cjs/loader:1198:14)
    at requireFromString (/home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/.pnpm/require-from-string@2.0.2/node_modules/require-from-string/index.js:28:4)
    at /home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/.pnpm/danger@11.3.0_@octokit+core@3.6.0/node_modules/danger/distribution/runner/runners/inline.js:161:68
    at step (/home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/.pnpm/danger@11.3.0_@octokit+core@3.6.0/node_modules/danger/distribution/runner/runners/inline.js:52:23)
    at Object.next (/home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/.pnpm/danger@11.3.0_@octokit+core@3.6.0/node_modules/danger/distribution/runner/runners/inline.js:33:53)
```
### Dangerfile
```
------^
```
  
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0372bffd4d117204ce1f0543b59c15d5f4f8b4d1
</p>

File: types/eslint-ast-utils/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
+// Type definitions for eslint-ast-utils 1.1
+// Project: https://github.com/jfmengels/eslint-ast-utils#readme
+// Definitions by: Mark McCann <https://github.com/marksmccann>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
```
Comment (jakebailey) [inline types/eslint-ast-utils/index.d.ts:4]: This is still an old-style package; the CI apparently don't catch this yet (oops) but you will need a `package.json`. The new dts-gen release should generate this properly, or you can model another package.

File: types/eslint-ast-utils/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": true,
```
Comment (jakebailey) [inline types/eslint-ast-utils/tsconfig.json:14]: `esModuleInterop` should not appear on DT and should be removed. Your tests should write:

```ts
import astUtils = require("eslint-ast-utils");
```

To fix that failure you saw.

Also, trailing commas aren't allowed as this file must be valid JSON for the infra to be able to read it (whereas TS is less strict).

File: types/eslint-ast-utils/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
+import * as ESLint from "eslint";
+import * as ESTree from "estree";
+
+declare const astUtils: {
```
Comment (jakebailey) [inline types/eslint-ast-utils/index.d.ts:4]: The upstream package just uses a plain export object; these can just be top level `export function` declarations: https://unpkg.com/browse/eslint-ast-utils@1.1.0/index.js

The only reason to do an `export =` is if the upstream package does `module.exports` on something funky, like a function/class that has been modified with properties or something.

File: types/eslint-ast-utils/tsconfig.json (hunk 2 above)
Comment (marksmccann) [reply types/eslint-ast-utils/tsconfig.json]: I figured, I just wasn't sure how else to resolve it. Thanks!

File: types/eslint-ast-utils/index.d.ts (hunk 3 above)
Comment (marksmccann) [reply types/eslint-ast-utils/index.d.ts]: Gotcha, updated!

**Review by jakebailey — APPROVED** [review]

**Comment by marksmccann** [comment]
Ready to merge

**Pull request merged**